prov/util/src/util_mem_hooks.c

```diff
@@ -525,78 +542,55 @@ static int ofi_memhooks_start(struct ofi_mem_monitor *monitor)
 	for (i = 0; i < OFI_INTERCEPT_MAX; ++i)
 		dlist_init(&intercepts[i].dl_intercept_list);
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_DLOPEN],
-				   (void **) &real_calls.dlopen);
-	if (ret) {
-		FI_WARN(&core_prov, FI_LOG_MR,
-		       "intercept dlopen failed %d %s\n", ret, fi_strerror(ret));
-		return ret;
-	}
-
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MMAP],
-				   (void **) &real_calls.mmap);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MMAP]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept mmap failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MUNMAP],
-				   (void **) &real_calls.munmap);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MUNMAP]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept munmap failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MREMAP],
-				   (void **) &real_calls.mremap);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MREMAP]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept mremap failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MADVISE],
-				   (void **) &real_calls.madvise);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_MADVISE]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept madvise failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SHMAT],
-				   (void **) &real_calls.shmat);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SHMAT]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept shmat failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SHMDT],
-				   (void **) &real_calls.shmdt);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SHMDT]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept shmdt failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_BRK],
-				   (void **) &real_calls.brk);
+	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_BRK]);
 	if (ret) {
 		FI_WARN(&core_prov, FI_LOG_MR,
 		       "intercept brk failed %d %s\n", ret, fi_strerror(ret));
 		return ret;
 	}
 
-	ret = ofi_intercept_symbol(&intercepts[OFI_INTERCEPT_SBRK],
```

Comment:
sbrk not needed?  Perhaps a comment line indicating why that is?

Reply:
sbrk is not a syscall.  It uses brk underneath